run_tests.py

```diff
@@ -0,0 +1,52 @@
+import os
+import sys
+import contextlib
+import subprocess
+
+
+@contextlib.contextmanager
+def binding(binding):
+    """Prepare an environment for a specific binding"""
+
+    sys.stderr.write("""\
+#
+# Running tests with %s..
+#
+""" % binding)
+
+    os.environ["QT_PREFERRED_BINDING"] = binding
+
+    try:
+        yield
+    except:
+        pass
+
+    os.environ.pop("QT_PREFERRED_BINDING")
+
+
+if __name__ == "__main__":
+    argv = [
+        "nosetests",
+        "--verbose",
+        "--with-process-isolation",
+        "--exe",
+    ]
+
+    argv.extend(sys.argv[1:])
+
+    # Running each test independently via subprocess
+    # enables tests to filter out from tests.py before
+    # being split into individual processes via the
+    # --with-process-isolation feature of nose.
+    with binding("PyQt4"):
+        subprocess.call(argv)
```

Comment:
This could have been:

```python
subprocess.call(argv, env={"QT_PREFERRED_BINDING": "PyQt4"})
subprocess.call(argv, env={"QT_PREFERRED_BINDING": "PySide"})
subprocess.call(argv, env={"QT_PREFERRED_BINDING": "PySide2"})
subprocess.call(argv, env={"QT_PREFERRED_BINDING": "PyQt5"})
```

But I personally think the current look is more readable.